llvm/lib/Transforms/Utils/SimplifyCFG.cpp

```diff
@@ -3578,6 +3578,14 @@ static bool FoldTwoEntryPHINode(PHINode *PN, const TargetTransformInfo &TTI,
   return true;
 }
 
+static BranchInst *decayCondBranchToUncondBranch(IRBuilderBase &Builder,
+                                                 BranchInst *BI, bool Eval) {
+  unsigned SuccessorIdx = (Eval) ? 0 : 1;
+  auto *NewBI = Builder.CreateBr(BI->getSuccessor(SuccessorIdx));
+  BI->eraseFromParent();
```

Comment:
This fails to update the dominator tree. I think it would be better to just replace the branch condition with true/false and let ConstantFoldTerminator take care of the rest.